### PR TITLE
Update contributing docs with current practice

### PR DESCRIPTION
**Pull request by rmartin16**
## Changes
- Update contributing docs for current practice (mostly the dev process)
- Reflow (english) docs for 80 character width

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [X] I have read the **CONTRIBUTING.md** file
- [X] I will abide by the code of conduct

### CODE REVIEW

**Comment by rmartin16** [comment]
The primary goal of this PR was to update https://beeware.org/contributing/how/process/ so I can reference it in the implementation of https://github.com/beeware/.github/pull/63.

I also went through the rest of the pages under `contributing` to make sure they weren't especially out of date; other than reflowing the text, I updated some punctuation and removed references to Travis and BeeKeeper.

**Review by freakboy3742 — COMMENTED** [review]
On the whole, these all look like good (and *badly* needed) changes. I initially started with a pure review, but some of the changes needed to be a bit more substantive, so I've pushed some updates:

* Removed all references to DCOs. We're not using or enforcing them, and I'm not convinced they're actually needed.
* Expanded the section on change notes to provide a practical example
* Added a section on translating the tutorial.

Marking as approved from my perspective; if you spot any typos or inconsistencies in what I've pushed, let me know. Otherwise, I'm happy to have this land as is.

File: content/contributing/how/translations/contents.lr [hunk 1]
```diff
@@ -1,5 +1,7 @@
 _model: page
 ---
+_template: translations.html
+---
```
Comment (freakboy3742) [inline content/contributing/how/translations/contents.lr:4]: This removal was actually deliberate - the translations template only serves to add the list of translated languages to the body of the text. In the previous version, the body ended with a "We have translations for:" line, which flowed into this extra content; but with the new body content, it makes less sense.

I've removed the template from the translated versions of this page as well; so - we either need to restore it to all versions, or delete it everywhere and delete the template entirely.

Comment (rmartin16) [reply content/contributing/how/translations/contents.lr]: I see. Are these translations discoverable at all without those links from the translations template?

I updated the wording....but it's just a strange instruction to "find the page in the language" you want to update...

Comment (rmartin16) [reply content/contributing/how/translations/contents.lr]: ahh...there's a globe next to the Edit button.

Comment (freakboy3742) [reply content/contributing/how/translations/contents.lr]: Yes - there's a pulldown at the top of the page (the globe icon) that lets you select one of the existing translations.

Comment (rmartin16) [reply content/contributing/how/translations/contents.lr]: Ok; Update the instructions to use the globe to choose a language prior to clicking edit.

**Review by freakboy3742 — APPROVED** [review]
Looks good to me! Thanks for prodding this forward!

**Pull request merged**